plugins/tiddlywiki/multiwikiserver/modules/command-mws-load-tiddlers.js

```diff
@@ -0,0 +1,40 @@
+/*\
+title: $:/plugins/tiddlywiki/multiwikiserver/mws-load-tiddlers.js
+type: application/javascript
+module-type: command
+
+Command to recursively load a directory of tiddler files into a bag
+
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+exports.info = {
+	name: "mws-load-tiddlers",
+	synchronous: true
+};
+
+var Command = function(params,commander,callback) {
+	this.params = params;
+	this.commander = commander;
+	this.callback = callback;
+};
+
+Command.prototype.execute = function() {
+	var self = this;
+	// Check parameters
+	if(this.params.length < 2) {
+		return "Missing pathname and/or bagname";
+	}
+	var tiddlersPath = this.params[0],
+		bagName = this.params[1];
+	$tw.mws.store.saveTiddlersFromPath(tiddlersPath,bagName);
```

Comment:
I think I can directly call this method, after a git sync from remote is done.

And call "save tiddlers" when user want to sync To remote. What is differences between load-archive and load-tiddlers? And seems there is no save-tiddlers?

Reply:
> I think I can directly call this method, after a git sync from remote is done. And call "save tiddlers" when the user wants to sync To remote, or every 30min.

It is my intention provide an action widget equivalent of this command, so that we will be able to extend the new "filtered actions" mechanism in #7999 to trigger saving the archive at intervals if there have been changes.

> What are the differences between load-archive and load-tiddlers? And it seems there are no save-tiddlers?

`mws-load` and `mws-save` introduce a standard archive format for MWS, with a JSON file for each tiddler, recipe and bag. The `load-tiddlers` command loads a bag with the tiddlers from a tiddler folder, which is an ordinary folder containing `.tid`, `.multids`, `tiddlywiki.files` etc, the same as the "tiddlers" subfolder of a wiki folder.

I've no objection to a `save-tiddlers` command that saves tiddlers as `.tid` files. However, until we extend the format to allow multiline fields, some tiddlers would presumably end up as `.json` files.

I specifically want to avoid reusing our existing mechanism for writing changes back to a wiki folder. There are many edge cases that make it unreliable and brittle (for example, editing a tiddler from a `.multids` file).

Reply:
This is OK, looks like `load-tiddlers` and (hopefully there can be a) `save-tiddlers` are compatible to the old nodejs folder, which is being used in the client side. And `mws-load` `mws-save` can be used in the Real server side.

`.multids` won't be a problem because most of client user won't touch the folder under the hood, they only use wiki UI. Same as multiline fields, it only generated by very few plugin, as long as not all tiddler become JSON, I think user will accept some tiddler become json.